client/src/modules/Course/contexts/SessionContext.tsx

```diff
@@ -8,7 +8,12 @@ import Router from 'next/router';
 import React, { useEffect } from 'react';
 import { useAsync } from 'react-use';
 
-export const SessionContext = React.createContext<Session>({} as Session);
+type DefaultContext = {
+  session: Session;
+  activeCourse?: ProfileCourseDto;
```

Comment:
doesn't it make sense to separate. ie Session and CourseContext are different to me. Not all pages have course, so I would split to Session & Course. Plus we have this `withCourseData`, which could be eventually transformed to context provider as well. in this case you won't need to mix up these things